airflow-core/docs/howto/deadline-alerts.rst

```diff
@@ -0,0 +1,255 @@
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Deadline Alerts
+===============
+
+The :class:`~airflow.sdk.definitions.deadline.DeadlineAlert` feature is the next evolution of
+the old SLA.  Deadline Alerts allow you to set time thresholds for your DAG runs and automatically
+respond when those thresholds are exceeded. You can set up Deadline Alerts by choosing a built-in
+reference point, setting an interval, and defining a response using either Airflow's Notifiers or
+a custom callback function.
+
+Creating a Deadline Alert
+-------------------------
+
+To create a Deadline Alert, you'll need to specify three components:
+
+* A reference: When to start counting from
+* An interval: How far before or after the reference point to trigger the alert
+* A callback: What to do when the deadline is exceeded
+
+Here is how Deadlines are calculated:
+
+::
+
+    [Reference] ------ [Interval] ------> [Deadline]
+        ^                                     ^
+        |                                     |
+     Start time                          Trigger point
+
+Here's an example DAG implementation. If the DAG has not finished 15 minutes after it was queued, send an email:
+
+.. code-block:: python
+
+    from datetime import datetime, timedelta
+    from airflow import DAG
+    from airflow.sdk.definitions.deadline import DeadlineAlert, DeadlineReference
+    from airflow.providers.smtp.notifications.smtp import SmtpNotifier
+    from airflow.providers.standard.operators.empty import EmptyOperator
+
+    with DAG(
+        dag_id="deadline_alert_example",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=15),
+            callback=SmtpNotifier(
+                to="team@example.com",
+                subject="[Alert] DAG {{ dag.dag_id }} exceeded time threshold",
+                html_content="The DAG has been running for more than 15 minutes since being queued.",
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+The timeline for this example would look like this:
+
+::
+
+    |------|-----------|---------|-----------|--------|
+        Scheduled    Queued    Started    Deadline
+         00:00       00:03      00:05      00:18
```

Comment:
It is 15 minutes right? Other than that, I love this section, it makes it super easy to understand the concept of deadlines!

```suggestion
         00:00       00:03      00:05      00:15
```

Reply:
Yup, but it's 15 minutes after the QUEUED_AT in this example, so 0:18 is right.  If you used the LOGICAL_DATE reference i would be at 0:15